### DISCUSSION TITLE
SD card error during serial monitor initialization

### DISCUSSION THREAD

**Discussion by Paar86** [sanni/cartreader#160] [Need help]
I have this weird issue with my SD card reader. Every time I start the serial monitor in Arduino app I get SD card initialization error with text "SD Error". I have built a custom Arduino Mega 2560 programmer with only support for 29F1610 EEPROMs and used pinout.xls file located in Git repository as reference. Pins for the SD card are as followed:

MISO - D50
MOSI - D51
SCK - D52
SS (labeled as CS on my board) - D52

Am I doing something wrong? Is there something I can check to be sure it's wired correctly? Voltage and GND pins are soldered correctly. Voltage is wired to +5V rail. Would appreciate any help.

![20210528_162924](https://user-images.githubusercontent.com/34010694/120000160-d0ad7c80-bfd2-11eb-89aa-984a206dabe3.jpg)

**Comment by sanni** [comment]
What version of the Cart_Reader.ino are you using? Newer ones work best with the SD card formatted to exFAT.

**Reply by Paar86** [reply]
Originally I had it formatted to FAT32 with 16kB cluster size. Tried to format it to exFAT/32kB and then to exFAT/16kB, with no luck, still getting "SD Error" message. The Arduino itself looks fine, I can upload the code with no problem. SD card reader has only four wire connections and I double checked them all. I don't know what could be wrong.

**Comment by sanni** [comment]
Do the examples from the SdFat library work? Could just be that the SD card is not compatible or that the SD module is broken.

**Comment by Paar86** [comment]
I'm using 16GB Kingston Class 10 UHS mSD card. I'll try to source another one and see if it'll solve the problem. I'll try some things with the SDFat library but my experience with Arduino stuff is very limited. :/
I'll try to figure something out.

**Comment by sanni** [comment]
In the Arduino IDE go File-> Examples -> SdFat(at the bottom of the menu) -> ExFatLogger
Upload to the Arduino, then open the serial monitor, type a character to start, then type l to list the files on the sd card.

If that doesn't work then something is wrong with either the Arduino, the SD card or the SD module.

You can also try a FAT32 formatted SD card with one of the earlier releases of the Cart_Reader.ino from last year. They still use the old SdFat library that might be more compatible.

**Reply by Paar86** [reply]
Tried the steps you've recommended with the original exFAT card and with another FAT32 card. No success. I'll try to desolder the reader from my board and connect it directly to the Arduino and if that won't help, will try to order new reader modules. Thanks for the tips!

**Comment by Paar86** [comment]
Hi! I would like to confirm that it was indeed a faulty SD card adapter. Replaced it with another one and the programmer works like charm. Thank you for your amazing work!

**Reply by Dakkaron** [reply]
I had a similar problem. The reason was a faulty solder point that resulted in no power to the sd reader
